examples/traefik/docker-compose.yml

```diff
@@ -167,3 +167,32 @@ services:
       - "REDIS_DB=0"
       - "SIDECAR_SYSLOGNG=1"
     restart: always
+
+  snmptrapd:
+    image: librenms/librenms:latest
+    container_name: librenms_snmptrapd
+    hostname: librenms-snmptrapd
+    depends_on:
+      - librenms
+    ports:
+      - target: 162
+        published: 162
```

Comment:
I have tried this PR with some local changes, snmptraps are not seen in the LibreNMS, because of the docker networking.

By default docker uses `bridge` network mode and translated the source IP for these published ports (`DNAT`), so, `snmptrapd` container does not know which host has sent this (both hostname and IP address are translated), so, it just ignores the trap.

I have tried several combinations of networking but none of them works. 

E.g., using `host` mode networking does not allow containers to talk to each other
E.g. tried some custom networking options, but disabling source address translation for NAT doesn't make sense.

So, we need a networking scheme with the below properties, but there is no such mode in docker AFAIK (I am still looking at custom iptables for docker).

* Bridge all containers
* Published ports for `snmptrapd` container should act like `host` mode or at least no NATing.

And we cannot even workaround this problem as the source information is lost, only way I can see now is to run all of them in a single container.

Reply:
Well, I found a working way with `host` networking driver in docker with the below patch, earlier I was trying with `localhost` and MySQL was suing UNIX socket by default as they are in separate containers it was failing, I have modified to use IP address variant of `localhost` and it worked.

```
diff --git a/docker-compose.yml b/docker-compose.yml
index 6b210fc..8fb1828 100644
--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -4,6 +4,7 @@ services:
   db:
     image: mariadb:10.4
     container_name: librenms_db
+    network_mode: host
     command:
       - "mysqld"
       - "--innodb-file-per-table=1"
@@ -23,6 +24,7 @@ services:
   memcached:
     image: memcached:alpine
     container_name: librenms_memcached
+    network_mode: host
     environment:
       - "TZ=${TZ}"
     restart: always
@@ -30,6 +32,7 @@ services:
   redis:
     image: redis:5.0-alpine
     container_name: librenms_redis
+    network_mode: host
     environment:
       - "TZ=${TZ}"
     restart: always
@@ -37,6 +40,7 @@ services:
   msmtpd:
     image: crazymax/msmtpd:latest
     container_name: librenms_msmtpd
+    network_mode: host
     env_file:
       - "./msmtpd.env"
     restart: always
@@ -44,6 +48,7 @@ services:
   librenms:
     image: /:latest
     container_name: librenms
+    network_mode: host
     hostname: librenms
     cap_add:
       - NET_ADMIN
@@ -64,7 +69,7 @@ services:
       - "TZ=${TZ}"
       - "PUID=${PUID}"
       - "PGID=${PGID}"
-      - "DB_HOST=db"
+      - "DB_HOST=127.0.0.1"
       - "DB_NAME=${MYSQL_DATABASE}"
       - "DB_USER=${MYSQL_USER}"
       - "DB_PASSWORD=${MYSQL_PASSWORD}"
@@ -74,6 +79,7 @@ services:
   dispatcher:
     image: /:latest
     container_name: librenms_dispatcher
+    network_mode: host
     hostname: librenms-dispatcher
     cap_add:
       - NET_ADMIN
@@ -89,7 +95,7 @@ services:
       - "TZ=${TZ}"
       - "PUID=${PUID}"
       - "PGID=${PGID}"
-      - "DB_HOST=db"
+      - "DB_HOST=127.0.0.1"
       - "DB_NAME=${MYSQL_DATABASE}"
       - "DB_USER=${MYSQL_USER}"
       - "DB_PASSWORD=${MYSQL_PASSWORD}"
@@ -104,6 +110,7 @@ services:
   syslogng:
     image: /:latest
     container_name: librenms_syslogng
+    network_mode: host
     hostname: librenms-syslogng
     cap_add:
       - NET_ADMIN
@@ -125,7 +132,7 @@ services:
       - "TZ=${TZ}"
       - "PUID=${PUID}"
       - "PGID=${PGID}"
-      - "DB_HOST=db"
+      - "DB_HOST=127.0.0.1"
       - "DB_NAME=${MYSQL_DATABASE}"
       - "DB_USER=${MYSQL_USER}"
       - "DB_PASSWORD=${MYSQL_PASSWORD}"
@@ -136,6 +143,7 @@ services:
   snmptrapd:
     image: /:latest
     container_name: librenms_snmptrapd
+    network_mode: host
     hostname: librenms-snmptrapd
     cap_add:
       - NET_ADMIN
@@ -157,7 +165,7 @@ services:
       - "TZ=${TZ}"
       - "PUID=${PUID}"
       - "PGID=${PGID}"
-      - "DB_HOST=db"
+      - "DB_HOST=127.0.0.1"
       - "DB_NAME=${MYSQL_DATABASE}"
       - "DB_USER=${MYSQL_USER}"
       - "DB_PASSWORD=${MYSQL_PASSWORD}"
```

Reply:
Of course, this means that any conflicts with the host networking have to be resolved, but I guess there is no other way to get SNMP traps.